### PR TITLE
ContractSdk add built-in types

### LINKED ISSUE #190: ContractSDK add some built-in data types
**Brief of your idea**
Smart contract SDK needs to provide rich types to improve user usability.
Built-in data types contains block and transaction this time.

**What could be better if XuperChain using your idea**
Users can develop smart contracts more easily.

**Are there any side effects?**
Describe what might be worse.

**Additional information**
If possible, include a minimal, complete, and verifiable example to help us identify the issue.

### PR DESCRIPTION
**Pull request by MichelleZ**
## Description

What is the purpose of the change?
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes #190 

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

## Brief of your solution
refer #190 

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

### CODE REVIEW

File: contractsdk/cpp/xchain/context_impl.cc [hunk 1]
```diff
@@ -85,6 +86,38 @@ bool ContextImpl::delete_object(const std::string& key) {
     return true;
 }
 
+bool ContextImpl::query_tx(const std::string &txid, Transaction* tx) {
+    pb::QueryTxRequest req;
+    pb::QueryTxResponse rep;
+    
+    std::string rawTxid = hex2string(txid);
```
Comment (icexin) [inline contractsdk/cpp/xchain/context_impl.cc]: Here pass the hex id, decode in the bridge

File: contract/bridge/syscall_service.go [hunk 2]
```diff
@@ -45,12 +45,27 @@ func (c *SyscallService) QueryBlock(ctx context.Context, in *pb.QueryBlockReques
 		return nil, err
 	}
 
-	blockbuf, err := json.Marshal(block)
-	if err != nil {
-		return nil, err
+	transactions := []*pb.Transaction{}
+	for _, t := range block.Transactions {
+		transaction := ConvertTxToSDKTx(t)
+		transactions = append(transactions, transaction)
```
Comment (icexin) [inline contract/bridge/syscall_service.go]: Since sdk only uses txid, there is no need to pass the whole tx here.

Comment (MichelleZ) [reply contract/bridge/syscall_service.go]: Shall we use original txid or hextxid stored here?

File: contract/bridge/syscall_service.go [hunk 3]
```diff
@@ -40,28 +41,28 @@ func (c *SyscallService) QueryBlock(ctx context.Context, in *pb.QueryBlockReques
 		return nil, fmt.Errorf("bad ctx id:%d", in.Header.Ctxid)
 	}
 
-	block, err := nctx.Cache.QueryBlock(in.Blockid)
+	rawBlockid, err := hex.DecodeString(in.Blockid)
```
Comment (icexin) [inline contract/bridge/syscall_service.go]: Error check?

Comment (MichelleZ) [reply contract/bridge/syscall_service.go]: Fix it

**Pull request merged**